src/main/java/duke/TasksList.java

```diff
@@ -0,0 +1,166 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.ToDo;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.io.FileWriter;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+public class TasksList {
+    private static final int LENGTH_OF_AT = 4;
+    private static final int LENGTH_OF_BY = 4;
+    private static final int LENGTH_OF_TODO = 5;
+    private static final int LENGTH_OF_EVENT = 6;
+    private static final int LENGTH_OF_DEADLINE = 9;
+    private static final String FILE_DIR = "./data";
+    private static final String FILE_NAME = "duke.txt";
+
+    protected ArrayList<Task> tasks = new ArrayList<>();
+
+    public void initDataStore() throws IOException {
+        File data_dir = new File(FILE_DIR);
+        if (data_dir.mkdir()) {
+            System.out.println("Directory " + FILE_DIR + " created");
+        }
+        File f = new File(FILE_DIR + "/" + FILE_NAME);
+        if (f.createNewFile()) {
+            System.out.println("File " + FILE_NAME + " created");
+        }
+    }
+
+    // Loads tasks that are stored in ./data/duke.txt
+    public void loadTasks() throws FileNotFoundException, DukeException{
+        File file = new File(FILE_DIR + "/" + FILE_NAME);
+        Scanner file_scanner = new Scanner(file);
+        while (file_scanner.hasNext()) {
+            Task task;
+            String raw_task_entry = file_scanner.nextLine();
+            String[] task_data_points = raw_task_entry.split("\\|");
+            switch (task_data_points[0]) {
+            case "T":
+                task = new ToDo(task_data_points[2]);
+                if (task_data_points[1].equals("1")) {
```

Comment:
Avoid using magic literals, considering putting the number into a variable with a meaningful name.